### ISSUE TITLE
App hangs after RTSP playback

### ISSUE THREAD

**Issue by chetan-prime** [valbok/QtAVPlayer#277]
Platform: Android
Qt: 6.3

My app uses QtAVPlayer within a QML form. Normal operation is when user presses the back button on QML, the form is changed.
This causes QtAVPlayer to call terminate() method etc. to perform usual cleanup. When playing back SRT source, there is no problem and the app continues to work fine. However with an RTSP source, the app hangs completely after the form changes.
In tracing I found the app terminate() function was hanging waiting for the RTSP demuxer to return. I then placed this code within the terminate() function:

demuxer.abort(true);

After this, the terminate function seems to cleanly return, however the app still hangs, as if there is some mutex stuck in locked state. I put in a lot of debug logs and even enabled qCDebug function, but am unable to trace where the lockup occurs. The strange thing this is it affects only RTSP sources so far. It hangs whether playback starts or the RTSP source is waiting for a stream.

**Comment by valbok** [comment]
Hi, thanks for the issue.
Do you have sample rtsp uri? is it for any uris? Can you check if it is relevant for your desktop?

**Comment by chetan-prime** [comment]
I was able to skip the hang by changing the parent component of the QtAVPlayer from a stub component of mine I had created with C++ that itself creates a child QAVPlayer.
I'm not sure if this actually fixes the issue, or is a way of hiding / working around the actual issue. Will check more and report back with some example code if I confirm the issue is with the ffmpeg related code. I did try switching ffmpeg versions from 4.5.1 -> 5.1 to no avail.
